heat/core/dndarray.py

```diff
@@ -45,13 +45,15 @@ def __setitem__(self, key, value):
 
 
 class DNDarray:
-    def __init__(self, array, gshape, dtype, split, device, comm):
+    def __init__(self, array, gshape, dtype, split, device, comm, halo=0):
```

Comment:
Would claim that their should be no explicit default here for halo. We need to make sure that the rest of the code is passing correct halo informations down internally.